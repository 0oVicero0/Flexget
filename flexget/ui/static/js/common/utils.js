'use strict';

String.prototype.startsWith = function(prefix) {
    return this.indexOf(prefix) === 0;
};

function registerFlexModule(module) {
  angular.module('flexgetApp').requires.push(module.name);
}