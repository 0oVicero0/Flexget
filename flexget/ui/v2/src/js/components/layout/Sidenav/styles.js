import styled, { css } from 'styled-components';
import theme from 'theme';
import Paper from 'material-ui/Paper';
import List, { ListItem } from 'material-ui/List';
import Button from 'material-ui/Button';

export const colorClass = css`color: ${theme.palette.accent[200]};`;

export const ListWrapper = styled(Paper)`
  background-color: ${theme.palette.accent[900]};
  transition: ${theme.transitions.create(['width', 'visibility'])};
  height: auto;
  width: ${({ open }) => (open ? '100vw' : 0)};
  border-radius: 0;
  display: flex;
  flex-direction: column;
  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};

  ${theme.breakpoints.up('sm')} {
    height: calc(100% - 5rem);
    width: ${({ open }) => (open ? '19rem' : '5rem')};
    position: fixed;
    visibility: visible;
  }
`;

export const Label = styled.p`
  composes: ${colorClass};
  flex: 1;
  text-transform: none;
  visibility: ${({ open }) => (open ? 'visible' : 'hidden')};
  opacity: ${({ open }) => (open ? 1 : 0)};
  transition: ${theme.transitions.create(['visibility', 'opacity'])};
  width: ${({ open }) => (open ? 'auto' : 0)};
`;

export const NavList = styled(List)`
  flex: 1;
`;

export const NavItem = styled(ListItem)`
  padding: 0;
  height: 4.8rem;
  border-left: 3px solid transparent;

  &:hover {
    border-left: 3px solid ${theme.palette.primary[500]};
  }
`;

export const NavButton = styled(Button)`
  display: flex;
  justify-content: center;
  width: 100%;
  height: 100%;
  min-width: 5rem;
  padding-top: 0;
  padding-bottom: 0;
`;

export const NavVersion = styled.div`
  display: ${({ hide }) => (hide ? 'none' : 'block')};
`;
