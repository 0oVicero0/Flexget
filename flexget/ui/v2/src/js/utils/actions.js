export const createAction = namespace => (type, payload, {
  ignore = false,
  message,
} = {}) => ({
  type,
  payload,
  error: (payload instanceof Error),
  meta: {
    ignore,
    message,
    namespace,
  },
});

export const createLoading = namespace => type => ({
  type,
  meta: {
    namespace,
    loading: true,
  },
});
