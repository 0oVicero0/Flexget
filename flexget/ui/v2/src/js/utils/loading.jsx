import Loadable from 'react-loadable';
import Loading from 'components/common/Loader';

export function createAsyncComponent(loader) {
  return Loadable({
    loading: Loading,
    loader,
  });
}

