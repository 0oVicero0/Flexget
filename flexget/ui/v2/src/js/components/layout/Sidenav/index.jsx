import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import classNames from 'classnames';
import { withStyles, createStyleSheet } from 'material-ui/styles';
import Button from 'material-ui/Button';
import List, { ListItem } from 'material-ui/List';
import Icon from 'material-ui/Icon';
import Paper from 'material-ui/Paper';
import Version from 'containers/layout/Version';
import 'font-awesome/css/font-awesome.css';

const styleSheet = createStyleSheet('SideNav', theme => ({
  listWrapper: {
    backgroundColor: theme.palette.accent[900],
    transition: theme.transitions.create(['width', 'visibility']),
    height: '100%',
    width: 190,
    borderRadius: 0,
    display: 'flex',
    flex: {
      direction: 'column',
    },
    [theme.breakpoints.up('sm')]: {
      width: 190,
    },
  },
  sideNavMini: {
    width: 0,
    visibility: 'hidden',
    [theme.breakpoints.up('sm')]: {
      visibility: 'visible',
      width: 50,
    },
  },
  label: {
    color: theme.palette.accent[200],
    flex: 1,
    textTransform: 'none',
  },
  labelMini: {
    display: 'none',
  },
  icon: {
    color: theme.palette.accent[200],
  },
  list: {
    flex: {
      grow: 1,
    },
  },
  listItem: {
    padding: 0,
    height: 48,
    borderLeftWidth: 3,
    borderLeft: 'solid transparent',
    '&:hover': {
      borderLeftWidth: 3,
      borderLeft: `solid ${theme.palette.primary[500]}`,
    },
  },
  button: {
    display: 'flex',
    justifyContent: 'center',
    width: '100%',
    height: '100%',
    minWidth: 50,
  },
  versionHide: {
    display: 'none',
  },
}));

const sideNavItems = [
  {
    link: '/log',
    icon: 'file-text-o',
    label: 'Log',
  },
  {
    link: '/execute',
    icon: 'cog',
    label: 'Execute',
  },
  {
    link: '/config',
    icon: 'pencil',
    label: 'Config',
  },
  {
    link: '/history',
    icon: 'history',
    label: 'History',
  },
  {
    link: '/movies',
    icon: 'film',
    label: 'Movies',
  },
  {
    link: '/pending',
    icon: 'check',
    label: 'Pending',
  },
  {
    link: '/schedule',
    icon: 'calendar',
    label: 'Schedule',
  },
  {
    link: '/seen',
    icon: 'eye',
    label: 'Seen',
  },
  {
    link: '/series',
    icon: 'tv',
    label: 'Series',
  },
  {
    link: '/status',
    icon: 'heartbeat',
    label: 'Status',
  },
];

class SideNav extends Component {
  static propTypes = {
    classes: PropTypes.object.isRequired,
    sideBarOpen: PropTypes.bool.isRequired,
  };

  renderNavItems() {
    const { classes, sideBarOpen } = this.props;
    return sideNavItems.map(({ link, icon, label }) => (
      <Link to={link} key={link}>
        <ListItem className={classes.listItem}>
          <Button color="accent" className={classes.button}>
            <Icon className={classNames('fa', `fa-${icon}`, classes.icon)} />
            <p className={
              classNames(
                classes.label,
                { [classes.labelMini]: !sideBarOpen },
              )}
            >
              {label}
            </p>
          </Button>
        </ListItem>
      </Link>
    ));
  }

  render() {
    const { classes, sideBarOpen } = this.props;
    return (
      <Paper
        className={classNames(
          classes.listWrapper,
          { [classes.sideNavMini]: !sideBarOpen },
        )}
        elevation={16}
      >
        <List className={classes.list}>
          { ::this.renderNavItems() }
        </List>
        <div className={classNames({ [classes.versionHide]: !sideBarOpen })}>
          <Version />
        </div>
      </Paper>
    );
  }
}

export default withStyles(styleSheet)(SideNav);
