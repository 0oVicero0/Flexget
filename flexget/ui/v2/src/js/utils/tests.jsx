import React from 'react';
import { MuiThemeProvider } from 'material-ui/styles';
import { MemoryRouter } from 'react-router-dom';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';

const mockStore = configureMockStore();

export function themed(component) {
  return (
    <MuiThemeProvider>
      { component }
    </MuiThemeProvider>
  );
}

export function router(component) {
  return (
    <MemoryRouter>
      { component }
    </MemoryRouter>
  );
}

export function provider(component, state = {}) {
  return (
    <Provider store={mockStore(state)}>
      { component }
    </Provider>
  );
}
