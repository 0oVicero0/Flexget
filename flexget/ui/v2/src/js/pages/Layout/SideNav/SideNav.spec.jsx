import React from 'react';
import renderer from 'react-test-renderer';
import Sidenav from 'pages/Layout/Sidenav';
import { provider, themed, router } from 'utils/tests';
import fetchMock from 'fetch-mock';

describe('pages/Layout/Sidenav', () => {
  beforeEach(() => {
    fetchMock
      .get('/api/server/version', {});
  });

  it('renders correctly with sideBarOpen', () => {
    const tree = renderer.create(
      provider(router(themed(<Sidenav sideBarOpen />)), { version: {} }),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('adds mini classes without sideBarOpen', () => {
    const tree = renderer.create(
      provider(router(themed(<Sidenav sideBarOpen={false} />)), { version: {} }),
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
