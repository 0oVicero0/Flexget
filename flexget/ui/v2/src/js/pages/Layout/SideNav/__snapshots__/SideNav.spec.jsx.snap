// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/Layout/Sidenav adds mini classes without sideBarOpen 1`] = `
.glamor-1 {
  color: #b0bec5;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-51 {
  margin: 0px;
}

.glamor-61 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 0;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-right: none !important;
}

.glamor-58 {
  color: #f44336;
  display: none;
}

@media (min-width:600px) {
  .glamor-61 {
    height: calc(100vh - 5rem);
    width: 6rem;
  }
}

<div
  className="MuiDrawer-docked-6"
>
  .glamor-1 {
  color: #b0bec5;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-51 {
  margin: 0px;
}

.glamor-61 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 0;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-right: none !important;
}

.glamor-58 {
  color: #f44336;
  display: none;
}

@media (min-width:600px) {
  .glamor-61 {
    height: calc(100vh - 5rem);
    width: 6rem;
  }
}

<div
    className="MuiPaper-root-8 MuiPaper-shadow0-10 MuiDrawer-paper-1 glamor-61 MuiDrawer-anchorLeft-2"
  >
    .glamor-1 {
  color: #b0bec5;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-51 {
  margin: 0px;
}

.glamor-58 {
  color: #f44336;
  display: none;
}

<div
      className="glamor-59 glamor-60"
    >
      .glamor-1 {
  color: #b0bec5;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<ul
        className="MuiList-root-35 MuiList-padding-36 glamor-48 glamor-49"
      >
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/config"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-pencil glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Config
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/history"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-history glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              History
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/log"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-book glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Log
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/movies"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-film glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Movies
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/pending"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-check glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Pending
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/seen"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-eye glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Seen
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/series"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-tv glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Series
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
          className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
          disabled={false}
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

<span
            aria-hidden="true"
            className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-tasks glamor-1"
          />
          .glamor-1 {
  color: #b0bec5;
}

<div
            className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
          >
            Tasks
          </div>
        </li>
      </ul>
      .glamor-1 {
  margin: 0px;
}

.glamor-8 {
  color: #f44336;
  display: none;
}

<div
        className="glamor-6 glamor-7 glamor-8"
      >
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          Version Info
        </p>
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          Flexget: 
           
           
        </p>
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          API: 
        </p>
      </div>
    </div>
  </div>
</div>
`;

exports[`pages/Layout/Sidenav renders correctly with sideBarOpen 1`] = `
.glamor-1 {
  color: #b0bec5;
}

.glamor-61 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 100vw;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-right: none !important;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-58 {
  color: #f44336;
  display: block;
}

.glamor-51 {
  margin: 0px;
}

@media (min-width:600px) {
  .glamor-61 {
    height: calc(100vh - 5rem);
    width: 19rem;
  }
}

<div
  className="MuiDrawer-docked-6"
>
  .glamor-1 {
  color: #b0bec5;
}

.glamor-61 {
  position: relative;
  background-color: #263238;
  height: calc(100vh - 10rem);
  width: 100vw;
  -webkit-transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  transition: width 300ms cubic-bezier(0.4, 0, 0.2, 1) 0ms;
  border-right: none !important;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-58 {
  color: #f44336;
  display: block;
}

.glamor-51 {
  margin: 0px;
}

@media (min-width:600px) {
  .glamor-61 {
    height: calc(100vh - 5rem);
    width: 19rem;
  }
}

<div
    className="MuiPaper-root-8 MuiPaper-shadow0-10 MuiDrawer-paper-1 glamor-61 MuiDrawer-anchorLeft-2"
  >
    .glamor-1 {
  color: #b0bec5;
}

.glamor-60 {
  width: inherit;
  height: inherit;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-box-pack: justify;
  -ms-flex-pack: justify;
  justify-content: space-between;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

.glamor-58 {
  color: #f44336;
  display: block;
}

.glamor-51 {
  margin: 0px;
}

<div
      className="glamor-59 glamor-60"
    >
      .glamor-1 {
  color: #b0bec5;
}

.glamor-49 {
  width: inherit;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<ul
        className="MuiList-root-35 MuiList-padding-36 glamor-48 glamor-49"
      >
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/config"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-pencil glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Config
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/history"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-history glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              History
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/log"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-book glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Log
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/movies"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-film glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Movies
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/pending"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-check glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Pending
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/seen"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-eye glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Seen
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<a
          href="/series"
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
            className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
            disabled={false}
            onClick={[Function]}
          >
            .glamor-1 {
  color: #b0bec5;
}

<span
              aria-hidden="true"
              className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-tv glamor-1"
            />
            .glamor-1 {
  color: #b0bec5;
}

<div
              className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
            >
              Series
            </div>
          </li>
        </a>
        .glamor-1 {
  color: #b0bec5;
}

.glamor-5 {
  border-left: 3px solid transparent;
  cursor: pointer;
}

.glamor-5:hover {
  border-left: 3px solid #ff9800;
}

<li
          className="MuiListItem-root-39 MuiListItem-gutters-46 MuiListItem-default-42 glamor-4 glamor-5"
          disabled={false}
          onClick={[Function]}
        >
          .glamor-1 {
  color: #b0bec5;
}

<span
            aria-hidden="true"
            className="material-icons MuiIcon-root-49 glamor-0 MuiListItemIcon-root-48 fa fa-tasks glamor-1"
          />
          .glamor-1 {
  color: #b0bec5;
}

<div
            className="MuiListItemText-root-56 MuiListItemText-inset-57 glamor-0 glamor-1"
          >
            Tasks
          </div>
        </li>
      </ul>
      .glamor-8 {
  color: #f44336;
  display: block;
}

.glamor-1 {
  margin: 0px;
}

<div
        className="glamor-6 glamor-7 glamor-8"
      >
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          Version Info
        </p>
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          Flexget: 
           
           
        </p>
        .glamor-1 {
  margin: 0px;
}

<p
          className="glamor-0 glamor-1"
        >
          API: 
        </p>
      </div>
    </div>
  </div>
</div>
`;
